src/data/sonata_data.cpp

```diff
@@ -114,6 +114,16 @@ void SonataData::record_data(double step, const std::vector<uint64_t>& node_ids)
     // Calculate the offset to write into the buffer
     uint32_t offset = static_cast<uint32_t>((step - last_step_recorded_) / reporting_period_);
     uint32_t local_position = last_position_ + total_elements_ * offset;
+
+    if (steps_recorded_ >= num_steps_) {
```

Comment:
We should use steps_to_write_ instead of num_steps_ here.

num_steps_ are the total steps of the simulation, while steps_to_write_ is the steps to be buffered before writing.

In our case probably it doesnt matter since the issue only appears when the duration is 0,025, so num_steps_ and steps_to_write will both be 1, but still I think is more correct.

Reply:
Ah ok! I didn't know this other variable that "tracks" counter within the "flush interval". 
I build+tested and looks fine as well.